src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -29,8 +31,15 @@ public static partial class Certificates
 
             private static X509Certificate2Collection GetCertificateCollection(string certificateFileName)
             {
+                // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
+                // This bug results in corrupting the X509Certificate2 certificate state.
+                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
+
                 var certCollection = new X509Certificate2Collection();
                 certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+
+                m.ReleaseMutex();
```

Comment:
With a machine-wide mutex, let's go out of our way to be as reliable as we can with releasing the mutex, e.g. doing releases in a finally.